internal/migration/migration.go

```diff
@@ -0,0 +1,478 @@
+package migration
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	json "github.com/nikkolasg/hexjson"
+	"go.etcd.io/bbolt"
+
+	"github.com/drand/drand/chain"
+	"github.com/drand/drand/chain/postgresdb/database"
+	"github.com/drand/drand/chain/postgresdb/pgdb"
+	"github.com/drand/drand/chain/postgresdb/schema"
+	"github.com/drand/drand/log"
+)
+
+type (
+	beacon struct {
+		PreviousSig []byte `json:",omitempty"`
+		Round       uint64
+		Signature   []byte
+	}
+
+	migrator struct {
+		ctx       context.Context
+		ctxCancel context.CancelFunc
+
+		startedAt time.Time
+		logger    log.Logger
+
+		pgDSN            string
+		bufferSize       int
+		beaconName       string
+		sourceBeaconPath string
+
+		// existingDB   *bbolt.DB
+		existingRows int
+
+		distChan    chan beacon
+		errChan     chan error
+		destination chain.StorageType
+	}
+)
+
+const ownerOnly = 0600
+
+const DefaultBufferSize = 10_000
+
+var (
+	// ErrMigrationNotNeeded ...
+	ErrMigrationNotNeeded = fmt.Errorf("migration not needed")
+
+	bucketName = []byte("beacons")
+)
+
+//nolint:lll // This function uses the right amount of params
+func Migrate(ctx context.Context, logger log.Logger, sourceBeaconPath, beaconName string, destination chain.StorageType, pgDSN string, bufferSize int) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
Assuming m.goMigrate is a blocking call, I don't understand the need for this context being created?